### PR TITLE
Keep empty object on JSON serialization

### LINKED ISSUE #1825: JSON inputs with empty or null values should be serialized
### Expected Behavior

_No response_

### Actual Behaviour

Following this PR https://github.com/kestra-io/kestra/pull/1674, we now serialize null / empty values of inputs. However when the input type is JSON, the actual input value is serialized by the default ObjectMapper which removes empty / null values.

See https://kestra-io.slack.com/archives/C03FEC452NQ/p1690446173575159

### Steps To Reproduce

_No response_

### Environment Information

- Kestra Version:
- Operating System (OS / Docker / Kubernetes):
- Java Version (If not docker):

### Example flow

_No response_

### LINKED ISSUE #2194: Fix serialization for empty and null values
Right now, we serialize all the data internally removing the `null` and empty (meaning empty array, empry object), see [here](https://github.com/kestra-io/kestra/blob/48edf70b382446c52c53e9a53884dd37f1104460/core/src/main/java/io/kestra/core/serializers/JacksonMapper.java#L117).

This lead to multiple issue and for now the response is not easy and muse be done mannually on every plugins : 
- https://github.com/kestra-io/plugin-aws/pull/260/commits/e96badc69e63641bfc766771883b3f5d6f682104 

This is already a first try [here](https://github.com/kestra-io/kestra/pull/1515) that show the change is not easy, since it was used everywhere (db, api, ...)

### LINKED ISSUE #3363: Return null if output cannot be retrieved instead of raising an error
### Feature description

If flow outputs cannot be generated due to some tasks being skipped, it would be better to return null instead of failing the execution. 

Reproducer - running with `false` will work fine. Change to `true` and then the output cannot be generated due to the task being skipped:

```yaml
id: myflow
namespace: test

inputs:
  - id: check
    type: BOOLEAN
    defaults: false

tasks:
  - id: downstream
    type: io.kestra.core.tasks.flows.If
    condition: "{{ inputs.check == true }}"
    then:
      - id: stop
        type: io.kestra.core.tasks.log.Log
        message: stopping
    else:
      - id: continue
        type: io.kestra.plugin.scripts.shell.Script
        runner: PROCESS
        script: |
          echo 1 > res.jsonl 
        outputFiles:
          - "*.jsonl"

outputs:
  - id: results
    type: JSON
    value: "{{ outputs.continue.outputFiles }}"
```

this execution will fail with input `true`

### PR DESCRIPTION
**Pull request by loicmathieu**
A second attempt at fixing [#2194](https://github.com/kestra-io/kestra/issues/2194) that have multiple implication.

Superseded https://github.com/kestra-io/kestra/pull/1515 (it take back its tests)   

Fixes #2194
Fixes #3363
Fixes #1825

### CODE REVIEW

**Comment by loicmathieu** [comment]
Note that `@JsonInclude(JsonInclude.Include.NON_DEFAULT)` is still present in the following classes otherwise, task defaults will not work properly: `Task`, `Input`, `AbstractTrigger`. To be coherent I also added the annotation to the model classes `Condition` and `TaskRunner` so task default can works there also.

File: core/src/test/java/io/kestra/core/runners/InputsTest.java [hunk 1]
```diff
@@ -316,4 +316,36 @@ void inputEnumFailed() {
...
+
+    @Test
+    void inputEmptyJson() {
+        HashMap<String, Object> map = new HashMap<>(inputs);
+        map.put("json", "{}");
+
+        Map<String, Object> typeds = typedInputs(map);
+
+        assertThat(typeds.get("json"), instanceOf(Map.class));
+        assertThat(((Map<?, ?>) typeds.get("json")).size(), is(0));
```
Comment (brian-mulier-p) [inline core/src/test/java/io/kestra/core/runners/InputsTest.java:334]: Please also add a test with `{"json":{"someEmptyObject":{}}}`

**Review by brian-mulier-p — APPROVED** [review]
I would say LGTM, just add the test I mentioned please

File: core/src/test/java/io/kestra/core/runners/InputsTest.java (hunk 1 above)
Comment (loicmathieu) [reply core/src/test/java/io/kestra/core/runners/InputsTest.java:334]: done

**Comment by loicmathieu** [comment]
@anna-geller some description for the BC

### Change in the way we serialize JSON objects

We previously serialized JSON objects with a `NON_DEFAULT` strategy; this means that only the properties that don't have default values will be included in the serialized JSON document. This was done to save space (in the database) and optimize network bandwidth.

We now switch to the more standard `NON_NULL` strategy, this means that all properties that are not null will be included in the serialized JSON document. This fixes some limitations and lets us know when a default is purposely set. This also means that empty lists and maps are now serialized as empty; previously, they were not serialized, so they are null, making using our Pebble expression template harder than necessary.

For example, if on a flow you had a single not mandatory `param` input and started your flow without passing a value, the expression `{{inputs.param}}` would have generated an error as the inputs list being empty it would not have been serialized due to the `NON_DEFAULT` strategy. With this change, the expression will no longer generate an error and will resolve to null.

**Pull request merged**